api/src/main/java/org/pmiops/workbench/firecloud/RawlsConfig.java

```diff
@@ -0,0 +1,124 @@
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.pmiops.workbench.auth.ServiceAccounts;
+import org.pmiops.workbench.auth.UserAuthentication;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.exceptions.ServerErrorException;
+import org.pmiops.workbench.rawls.ApiClient;
+import org.pmiops.workbench.rawls.api.BillingV2Api;
+import org.pmiops.workbench.rawls.api.WorkspacesApi;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ScopedProxyMode;
+import org.springframework.web.context.annotation.RequestScope;
+
+@org.springframework.context.annotation.Configuration
+public class RawlsConfig {
+  public static final List<String> BILLING_SCOPES =
+      ImmutableList.<String>builder()
+          .addAll(RawlsApiClientFactory.SCOPES)
+          .add("https://www.googleapis.com/auth/cloud-billing")
+          .build();
+
+  // Bean names used to differentiate between an API client authenticated as the end user (via
+  // UserAuthentication) and an API client authenticated as the service account user (via
+  // the service account access token).
+  //
+  // Some groups of FireCloud APIs will use one, while some will use the other.
+  //
+  public static final String END_USER_API_CLIENT = "rawlsEndUserApiClient";
+  public static final String END_USER_LENIENT_TIMEOUT_API_CLIENT =
+      "rawlsEndUserLenientTimeoutApiClient";
+  public static final String SERVICE_ACCOUNT_API_CLIENT = "rawlsServiceAccountApiClient";
+  public static final String SERVICE_ACCOUNT_WORKSPACE_API = "rawlsWorkspaceAclsApi";
+  public static final String END_USER_WORKSPACE_API = "rawlsWorkspacesApi";
+  public static final String END_USER_LENIENT_TIMEOUT_WORKSPACE_API =
+      "rawlsLenientTimeoutWorkspacesApi";
+  public static final String SERVICE_ACCOUNT_BILLING_V2_API = "serviceAccountBillingV2Api";
+  public static final String END_USER_STATIC_BILLING_V2_API = "endUserBillingV2Api";
+
+  @Bean(name = END_USER_API_CLIENT)
+  @RequestScope(proxyMode = ScopedProxyMode.DEFAULT)
+  public ApiClient endUserApiClient(
+      UserAuthentication userAuthentication, RawlsApiClientFactory factory) {
+    ApiClient apiClient = factory.newRawlsApiClient();
+    apiClient.setAccessToken(userAuthentication.getCredentials());
+    return apiClient;
+  }
+
+  @Bean(name = END_USER_LENIENT_TIMEOUT_API_CLIENT)
+  @RequestScope(proxyMode = ScopedProxyMode.DEFAULT)
+  public ApiClient endUserLenientTimeoutApiClient(
+      UserAuthentication userAuthentication,
+      RawlsApiClientFactory factory,
+      WorkbenchConfig config) {
+    ApiClient apiClient = factory.newRawlsApiClient();
+    apiClient.setAccessToken(userAuthentication.getCredentials());
+    apiClient
+        .getHttpClient()
+        .setReadTimeout(config.firecloud.lenientTimeoutInSeconds, TimeUnit.SECONDS);
```

Comment:
Workspace API 

> 



> Also, thank you for splitting this into another PR! I personally found this one much easier to review, since the complexity of the migrating all the endpoints is now isolated to the next PR.

NP! It is definitely to review, but it does add extra couple hours of splitting, rewriting some, then future solve conflift.
I underestimate the error so did not consider start from small. But actually I still prefer to keep that large PR since it is already there, and there is choice between 5+ extra hours of rewriting vs 1 extra hr reviewing code given the extra logic are really just fixing tests....

But anyway, it is good to hear spliting makes it easier to review.

Reply:
Fair, and I think this is still an improvement as-is 👍